pkg/arguments/arguments_test.go

```diff
@@ -0,0 +1,43 @@
+package arguments
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/spf13/cobra"
+)
+
+var _ = Describe("Client", func() {
+	var (
+		cmd *cobra.Command
+	)
+
+	Context("Test PreprocessUnknownFlagsWithId func", func() {
+		BeforeEach(func() {
+			cmd = &cobra.Command{
+				Use:   "test",
+				Short: "Test command used for testing non-positional args",
+				Long: "This test command is being used specifically for testing non-positional args, " +
+					"so we do not confuse users with hard rules for where, for example, the ID in " +
+					"`rosa edit addon ID` must be. For example, we want to be able to do `rosa edit addon " +
+					"-c test <ADDON_ID>` as well as `rosa edit addon <ADDON_ID> -c test`.",
+				Args: func(cmd *cobra.Command, argv []string) error {
+
+					return nil
+				},
+			}
+			cmd.Flags().BoolP("help", "h", false, "")
+		})
+		It("Returns without error", func() {
+			err := PreprocessUnknownFlagsWithId(cmd, []string{"test", "-c", "test-cluster"})
+			Expect(err).ToNot(HaveOccurred())
+		})
+		It("Returns error with no ID", func() {
+			err := PreprocessUnknownFlagsWithId(cmd, []string{"-c", "test-cluster"})
+			Expect(err).To(HaveOccurred())
+		})
+		It("Returns error with flag that has no value", func() {
+			err := PreprocessUnknownFlagsWithId(cmd, []string{"test", "-c", "-c", "-c", "-c"})
```

Comment:
@hunterkepley See my comment above about testing the actual error message that we have here as I think we might be logging out the incorrect argument.